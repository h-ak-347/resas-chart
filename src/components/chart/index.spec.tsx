global.ResizeObserver = require('resize-observer-polyfill');
import '@testing-library/jest-dom';
import Chart from './index.tsx';
import { render } from '@testing-library/react';

const mockDefaultProps: React.ComponentProps<typeof Chart> = {
  selectedPrefectures: [{ prefCode: 1, prefName: '北海道' }],
  chartData: [
    { year: 1960, 北海道: 1000 },
    { year: 1965, 北海道: 2000 },
    { year: 1970, 北海道: 1500 },
  ],
};

describe('Chart', () => {
  it('should match to snapshot.', async () => {
    const { asFragment } = render(<Chart {...mockDefaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
