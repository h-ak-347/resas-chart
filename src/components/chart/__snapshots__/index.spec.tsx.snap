// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should match to snapshot. 1`] = `
<DocumentFragment>
  <div
    class="chart"
  >
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 100%; min-width: 0;"
    />
  </div>
</DocumentFragment>
`;
